src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -93,23 +93,24 @@ function SubmissionsFactory(
     xhr.send(fd)
     // On Response
     xhr.onreadystatechange = function () {
-      // 4 DONE  The operation is complete.
       const OPERATION_DONE = 4
       if (xhr.readyState === OPERATION_DONE) {
         // waterfall is successful
-        if (xhr.status === HttpStatus.OK) {
-          deferred.resolve('Submission has finished.')
-        } else {
-          let response = {}
-          try {
-            response = JSON.parse(xhr.responseText)
-            // eslint-disable-next-line no-empty
-          } catch (e) {}
+        let response = {}
+        try {
+          response = JSON.parse(xhr.responseText)
+          if (xhr.status === HttpStatus.OK) {
+            deferred.resolve({
+              message: 'Submission has finished.',
+              submissionId: response.submissionId,
+            })
+          } else {
+            deferred.reject(`${response.message}`)
+          }
+          // eslint-disable-next-line no-empty
+        } catch (e) {
```

Comment:
nit: if this is unused, then we can omit. but should we default to the message on the error or this string?

Reply:
If I'm not wrong, I think this error message will be shown to the user so I think it would be better to keep this original error message to inform the user of possible actions they can take, rather than the actual error message that a user might not understand?

Reply:
oh just to be clear, i'm talking about the `e` parameter, which isn't used (i think), so maybe should remove!

Reply:
yeah I think you can just delete the `e`, so you end up with `} catch () {`

Reply:
I think removing the `e` causes an error (`Identifier expected`).

Reply:
from the [mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch), i think you can just omit `()` entirely, which is what is subpect is causing the error!
![Screenshot from 2021-05-18 17-39-09](https://user-images.githubusercontent.com/44049504/118629448-2a0df280-b800-11eb-9f03-88ea644b29e5.png)

Reply:
Eslint complains about this somehow 
<img width="945" alt="Screenshot 2021-05-19 at 11 08 53 AM" src="https://user-images.githubusercontent.com/54089291/118750756-9fc29e80-b892-11eb-8d69-df2f5a266359.png">

Reply:
nvm just keep the `e` lmao